### DISCUSSION TITLE
Arduino ide updated and now my code won't compile

### DISCUSSION THREAD

**Discussion by pb21uk** [espressif/arduino-esp32#10441] [Q&A]
I am using Arduino ide 2.3.3 with an esp32 plus board by Keyestudio. I have been creating code and this morning after the ide started and did an update to dependencies and all I am getting many compilation errors. I don't know what to do. Here is a small sample of the first few:
In file included from C:\Users\peter\AppData\Local\Arduino15\packages\esp32\hardware\esp32\3.0.5\libraries\Wire\src/Wire.h:39,
                 from C:\Users\peter\OneDrive\Documents\Arduino\Traffic_Lights_Fully_Working_copy_for_updates\Traffic_Lights_Fully_Working_copy_for_updates.ino:2:
C:\Users\peter\AppData\Local\Arduino15\packages\esp32\hardware\esp32\3.0.5\cores\esp32/HardwareI2C.h:24:1: error: expected unqualified-id before 'class'
   24 | class HardwareI2C : public Stream {
      | ^~~~~
C:\Users\peter\AppData\Local\Arduino15\packages\esp32\hardware\esp32\3.0.5\libraries\Wire\src/Wire.h:55:36: error: expected class-name before '{' token
   55 | class TwoWire : public HardwareI2C {
      |                                    ^
C:\Users\peter\AppData\Local\Arduino15\packages\esp32\hardware\esp32\3.0.5\libraries\Wire\src/Wire.h:92:8: error: 'bool TwoWire::begin()' marked 'override', but does not override
   92 |   bool begin() override final { 
Help would be very much appreciated and if I have posted incorrectly then please forgive me.

Regards in advance.

**Comment by me-no-dev** [comment]
@P-R-O-C-H-Y can you please give this a shot?

**Reply by pb21uk** [reply]
I am not sure how this works but are you guiding me to here.
![ide](https://github.com/user-attachments/assets/d1f217b7-9f9c-42c6-bc7b-8a3311d51829)
or something like this?
